### PR TITLE
Selectively enable "remove-vcs" instead of disabling interactivity

### LINKED ISSUE #285: Symfony Flex breaks interactivity of Composer scripts
Hi,

I have a composer script defined:

```json
{
    "scripts": {
        "post-install-cmd": [
            "My\\Cool\\ScriptHandler::run"
        ],
        "post-update-cmd": [
            "My\\Cool\\ScriptHandler::run"
        ]
    }
}
```

which works, except for one difference: the given InputStream is non-interactive.

```php
<?php

use Composer\Script\Event;

class ScriptHandler
{
    public function handle (Event $event)
    {
        var_dump($event->getIO()->isInteractive());
    }
}
```

The problem is flex, as this short snipped shows:

```
$ composer run-script parameters
> My\Cool\ScriptHandler::handle
bool(false)


$ composer remove symfony/flex
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 0 updates, 1 removal
  - Removing symfony/flex (v1.0.60)
[...]


$ composer run-script parameters
> My\Cool\ScriptHandler::handle
bool(true)
```

This currently makes implementing something similar to `incenteev/composer-parameter-handler` in combination with running flex impossible.

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixes #285, requires Composer v1.7.

### CODE REVIEW

File: src/Flex.php [hunk 1]
```diff
@@ -156,7 +157,11 @@ public function activate(Composer $composer, IOInterface $io)
             }
 
             if ('create-project' === $command) {
-                $input->setInteractive(false);
+                if (class_exists(Comparer::class)) {
```
Comment (maxhelias) [inline src/Flex.php]: The condition on the presence of the class is to determine if the option is present ?
If so, is not it be better to use the Composer constant like this `$this->composer::VERSION` ?

Comment (nicolas-grekas) [reply src/Flex.php]: Because the constant is not populated for snapshot versions of Composer...

Comment (nicolas-grekas) [reply src/Flex.php]: (comment added)

**Comment by apfelbox** [comment]
Thank you for taking this over @nicolas-grekas , patch looks good 👍

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**